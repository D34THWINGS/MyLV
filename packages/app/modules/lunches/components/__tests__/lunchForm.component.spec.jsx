import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { LunchForm } from '../lunchForm.component'

jest.unmock('../lunchForm.component')

describe('LunchForm', () => {
  let props

  const getWrapper = () => shallow(<LunchForm {...props} />)

  beforeEach(() => {
    props = {
      valid: true,
      pristine: true,
      handleSubmit: jest.fn(),
      render: jest.fn(({ children }) => <React.Fragment>{children}</React.Fragment>),
      classes: {},
    }
  })

  it('should render properly', () => {
    // When
    const wrapper = getWrapper()

    // Then
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
