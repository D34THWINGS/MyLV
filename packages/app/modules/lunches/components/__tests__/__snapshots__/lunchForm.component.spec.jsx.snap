// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LunchForm should render properly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <WithStyles(Grid)
    container={true}
  >
    <WithStyles(Grid)
      item={true}
      md={6}
      xs={12}
    >
      <Field
        component={[MockFunction]}
        fullWidth={true}
        label="IntitulÃ©"
        name="label"
        type="text"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      md={6}
      xs={12}
    >
      <Field
        component={[MockFunction]}
        fullWidth={true}
        label="Date"
        name="date"
        type="text"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <FieldArray
        component={[MockFunction]}
        name="attendees"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</form>
`;
